### DISCUSSION TITLE
SFTP failure to connect

### DISCUSSION THREAD

**Discussion by wnjohnson3** [iterate-ch/cyberduck#16374] [Q&A]
I’ve been using cyberduck for years but recently (new mac m1 sonoma) I can not use cyberduck to sftp connect to anything.  So I purchased 9.0.2 from the Apple Store and still having the same problem.

Most recent testing is connection to 2nd account on my mac.

command line sftp test@192.168.146.200 works just fine,

but doing the same with cyberduck gets Connection Failed:  Broken pipe. The connection attempt was rejected. The server may be down, or your network may not be properly configured.

1. Open Cyberduck browser
2. Click on test connection
3. Connection immediately fails and countdown 10 sec to retry
4. See Connection failed dialog box

**Expected behavior**
successful login to remote, and remote browser to to display directory
 
**Screenshots**
configuration
<img width="522" alt="Screenshot 2024-09-22 at 11 55 45 AM" src="https://github.com/user-attachments/assets/ad7cc160-8689-4804-bbd1-e6395377e853">

**Desktop (please complete the following information):**
 - OS: Mac OS Sonoma (M1 processor)
 - Version 14.7

**Log Files**
Please include the [log file]
[cyberduck.log](https://github.com/user-attachments/files/17090305/cyberduck.log)

**Additional context**

**Comment by vincentluciani** [comment]
Hello,
Is it related to the issue I raised (https://github.com/iterate-ch/cyberduck/issues/16366)?
Looking at your screenshot, I see the url (value in blue with hyperlink: sftp://192.168.146.200) does not contain the user name (should be sftp://test@192.168.146.200 ?). That would explain why the connection functioned properly with the command line (sftp test@192.168.146.200) versus the connection with Cyberduck (sftp 192.168.146.200).
I am also a Mac user

**Comment by wnjohnson3** [comment]
Yes, exactly.  I did not notice that.  In past versions the username was automatically added to the url.  Now I can’t figure out how to make that happen.

**Comment by tatze96** [comment]
> Now I can’t figure out how to make that happen.

You can't get Cyberduck to display the username within the URL, as it doesn't get displayed within the URL of the bookmark window on macOS, although it is included during the connection. see #16369

**Reply by vincentluciani** [reply]
Is it a coincidence that since then sftp does not function on Mac anymore but still functions on Windows?

**Comment by dkocher** [comment]
The log indicates a broken pipe error connecting to `192.168.146.200`.  Please share the verbose output for a successful connection with `ssh -v` from Terminal.app

**Reply by wnjohnson3** [reply]
different ip address (different location) but same computer/problem

[cli_ssh.log](https://github.com/user-attachments/files/17168097/cli_ssh.log)
[cyberduck_240923.log](https://github.com/user-attachments/files/17168098/cyberduck_240923.log)

**Reply by dkocher** [reply]
For some reason the server blocks the connection for Cyberduck. Check the server log for any related output.

```
Caused by: java.net.SocketException: Broken pipe
```

**Comment by wnjohnson3** [comment]
I include the command line ssh log and the attempted sftp connection via cyberduck.  Note the IP address has changed but the configuration and testing is the same.

[cli_ssh.log](https://github.com/user-attachments/files/17104783/cli_ssh.log)

[cyberduck_240923.log](https://github.com/user-attachments/files/17104847/cyberduck_240923.log)

**Reply by wnjohnson3** [reply]
I have two Macs.  
My home Mac has this problem.  It is M2 processor running Sonoma 14.7 and Cyberduck 9.0.2.
My work Mac executes the same tests without problems.  It is Intel processor running Sonoma 14.6.1 and Cyberduck 9.0.2.

Is this a M2 code bug???

**Reply by vincentluciani** [reply]
I have the same problem of connection. I have an M2 processor running no Sequoia 15.0, Cyberduck 9.0.2

**Reply by wnjohnson3** [reply]
I upgraded from Sonoma to Sequoia 15.0.1 today and my M2 MAC is now working.  My guess is the upgrade cleared up some trash that prevented cyberduck from working with Sonoma.

**Comment by tinohendricks** [comment]
I'm facing the same issue. Worked for years.
Any help in sight?

**Reply by dkocher** [reply]
Share a [log file](https://docs.cyberduck.io/cyberduck/support/#logging-output) and the error message.

**Reply by tinohendricks** [reply]
Ha! The culprit is the privacy protection setting for "local network" that obviously gets reset with (more or less) every macOS update.
Solved! Sorry for the noise.
